### PR TITLE
doc: fix incorrect example of `@statusCode`

### PR DESCRIPTION
**Pull request by mochi-yu**
## Issue Description
There is a example of `TypeSpec.http.@statusCode` [here](https://typespec.io/docs/next/libraries/http/reference/decorators#@TypeSpec.Http.statusCode). However, the property name is missing in this example. Due to this, the code did not work as expected.

```
op read(): {@statusCode: 200, @body pet: Pet}
op create(): {@statusCode: 201 | 202}
```

Instead, specifying the property name as follows made it work correctly:

```
op read(): {
  @statusCode _: 200;
  @body pet: Widget;
};
op create(): {
  @statusCode _: 201 | 202;
};
```

## Fix Description
- Corrected examples of `TypeSpec.http.@statusCode` in the reference and documentation.

### CODE REVIEW

**Comment by mochi-yu** [comment]
@microsoft-github-policy-service agree

**Comment by mochi-yu** [comment]
Dear Maintainer,

Thanks for this useful tool!
I was using TypeSpec and found this incorrect example. I wanted to contribute to this tool, so I committed this fix and created a pull request.
However, I do not have enough experience contributing to OSS projects. If I missed something, I would like you to fix it or support me.

File: packages/http/generated-defs/TypeSpec.Http.ts [hunk 1]
```diff
@@ -12,8 +12,13 @@ import type {
...
  * ```typespec
- * op read(): {@statusCode: 200, @body pet: Pet}
- * op create(): {@statusCode: 201 | 202}
+ * op read(): {
+ * @statusCode _: 200;
+ * @body pet: Pet;
+ * };
+ * op create(): {
+ * @statusCode _: 201 | 202;
+ * };
```
Comment (timotheeguerin) [inline packages/http/generated-defs/TypeSpec.Http.ts:21]:
```suggestion
 * op read(): {
 *   @statusCode _: 200;
 *   @body pet: Pet;
 * };
 * op create(): {
 *   @statusCode _: 201 | 202;
 * };
```

**Review by markcowl — APPROVED** [review]

**Pull request merged**